src/GafferImage/Warp.cpp

```diff
@@ -301,8 +301,8 @@ void Warp::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context
 		V2i tileOrigin = context->get<V2i>( ImagePlug::tileOriginContextName );
 		enginePlug()->hash( h );
 
-		filterOverDerivativesPlug()->hash( h );
-		if( filterOverDerivativesPlug()->getValue() )
+		useDerivativesPlug()->hash( h );
```

Comment:
Since hashing performance is pretty crucial, it's probably worth not calling `hash()` and then `getValue()` for the same plug here. We could just drop the `hash()` call and instead hash in the result of `getValue()`.